builder/source/simple.go

```diff
@@ -112,50 +113,52 @@ func (s *SimpleSource) IsFetched() bool {
 	return PathExists(s.GetPath(s.validator))
 }
 
-// download utilises CURL to do all downloads
+// download simple files using go grab
 func (s *SimpleSource) download(destination string) error {
-	hnd := curl.EasyInit()
-	defer hnd.Cleanup()
 
-	hnd.Setopt(curl.OPT_URL, s.URI)
-	hnd.Setopt(curl.OPT_FOLLOWLOCATION, 1)
+	file, err := os.Create(destination)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
 
-	out, err := os.Create(destination)
+	client := grab.NewClient()
+	req, err := grab.NewRequest(destination, s.URI)
 	if err != nil {
 		return err
 	}
 
-	pbar := pb.New64(0)
-	pbar.Set(pb.Bytes, true)
-	pbar.Set("prefix", filepath.Base(destination))
-	pbar.SetMaxWidth(80)
+	resp := client.Do(req)
 
-	writer := func(data []byte, udata interface{}) bool {
-		if _, err := out.Write(data); err != nil {
-			return false
+	// Setup our progress bar
+	var pbar *pb.ProgressBar
+	pbar = pb.Start64(resp.Size())
+	pbar.Set(pb.Bytes, true)
+	// Default template except that the 'per second' suffix is changed to '/s' instead of 'p/s'
+	progressBarTemplate := `{{with string . "prefix"}}{{.}} {{end}}{{counters . }} {{bar . }} {{percent . }} {{speed . "%s/s" "??/s"}}{{with string . "suffix"}} {{.}}{{end}}`
+	pbar.SetTemplateString(progressBarTemplate)
+	defer pbar.Finish()
+
+	// Timer to integrate into pbar
+	t := time.NewTicker(100 * time.Millisecond)
+	defer t.Stop()
+
+Loop:
+	for {
+		select {
+		case <-t.C:
+			pbar.SetCurrent(resp.BytesComplete())
+		case <-resp.Done:
+			break Loop
 		}
-		return true
 	}
-	progress := func(total, now, utotal, unow float64, udata interface{}) bool {
-		pbar.SetTotal(int64(total))
-		pbar.SetCurrent(int64(now))
 
-		return true
+	if err := resp.Err(); err != nil {
+		log.Errorf("Error downloading %s: %v\n", s.URI, err)
+		return err
 	}
 
-	hnd.Setopt(curl.OPT_WRITEFUNCTION, writer)
-	hnd.Setopt(curl.OPT_NOPROGRESS, false)
-	hnd.Setopt(curl.OPT_PROGRESSFUNCTION, progress)
-	// Enforce internal 300 second connect timeout in libcurl
-	hnd.Setopt(curl.OPT_CONNECTTIMEOUT, 0)
-	hnd.Setopt(curl.OPT_USERAGENT, fmt.Sprintf("solbuild 1.5.2.2"))
-
-	pbar.Start()
-	defer func() {
-		pbar.Finish()
-	}()
-
-	return hnd.Perform()
+	return resp.Err()
```

Comment:
I'd return `nil` here since you handled the error before, so that it;'s that if we got here everything went fine.